### ISSUE TITLE
Custom Fields for Non-Assets (Licenses, Accessories, Consumables, Components, Users, Models, etc)

### ISSUE THREAD

**Issue by snipe** [snipe/snipe-it#4261]
This has been proposed multiple times in various tickets, but I'm going to use this as a placeholder for that discussion moving forward, since the work behind extending custom fields beyond assets is basically the same, and would be a system-wide change. 

I'll be closing all of the other related tickets and directing everyone here moving forward, as this issue will be easier to search on since it contains everything.

**Comment by sjlaird** [comment]
I pulled this one out of #2411 so it doesn't get lost. I know it's talking about more than just custom fields. Do you want me to open a new issue for it?
> 
- It would also be great if there could be some history included that can be included for license renewal. At the moment we are thinking of using the comment field.

This has actually turned out to be an all important need for us. We need to track license purchases and renewals and maintenance. A software maintenance contract might be renewed on a yearly (or more) basis and might have a different Purchase Order number each time. It could have a different serial number (or even product name) so in fact, all the details may change, but the records all still need to be associated.
How about adding the ability to group individual license entries under a user provided name and providing a list of those groups.... For example, I create a license group called "ArcServe" in which I can see all of my Arcserve licnes under and I can sort by date... You probably have a better idea but I hope this explains the need.

**Comment by kagashe** [comment]
+1 for this would be great to have.

**Comment by tbradsha** [comment]
+1 as well. I was adding some chargers as accessories, and was hoping to have a volts/amps/connector fields so that I could track down compatible chargers as needed.

**Comment by Puddl36lum** [comment]
Any update on when these changes might get rolled out?

**Comment by snipe** [comment]
Not any time soon. It's enormously complicated to implement.

**Comment by TheVakman** [comment]
+1 I would also be very interested in this.

**Comment by neironus** [comment]
+1! Custom fields for components a very important feature.

**Comment by psgschladt** [comment]
+1 I would love to have my custom field for cable length of socket strips, network cables etc. Links to invoices on sharepoint etc.

**Comment by invuser** [comment]
To manage the life cycle of components e.g.  hard drives I am asking for more fields.
When the hard drive gets decommissioned, and it needs to be securely destroyed, the extra information is needed in the inventory database. ( if it was securely erased, when it was done, and info about physical destruction)
I consider H/D as a server COMPONENT of server ASSET. (component checked out to asset)
I could inventory  H/D as an asset, and create extra fields for h/d model asset, but I think it would be easier to treat H/D as a component with life cycle fields.
Thanks.

**Comment by patricmutwiri** [comment]
I'd love to have custom fields for non-assets also.

**Comment by harcesz** [comment]
+1

I'm adding mobile SIM, and memory cards as phone components, and need fields fore numbers, operators, storage and so on. 

Also; possibly a separate discussion, but blocking names for categories that are assigned only to a single type (asset or component or ...) is cosing problems for me, since for example "cables" might be assets, components or accessories, while still getting a full listing of every cable would be helpful.

**Comment by bdtsoccer** [comment]
+1.
Would really like to have this for components. My office is paying for hosted and overall we really like the system, but trying to keep track of component details is limited without having custom fields.
Ideally, we'd love to be able to have add more quantity to an existing component when we have a new order for the same component.
IE, we're upgrading PCs to SSDs, and we continually by new SSDs, mostly of the same model and capacity. Rather than having a separate component entry for every single order, I'd love to have a single Component entry that increases in quantity as we order more, and lists the separate order dates.

**Comment by esspeecy410** [comment]
+1
I'd love to see this on Companies and Locations. As an EPOS provider we are using Snipe-IT to record the various versions of our software amongst other things, and being able to record a Location or Company as a whole as running a particular version of our software by way of a Custom Field would be a great implementation for us.

**Comment by andreasoc** [comment]
custom fileds for license is very important! thanks

**Comment by kchagnon-snowbound** [comment]
Custom fields across the board is very important!

**Comment by batchn00b** [comment]
More Options on the Costs of licenses would be nice. Example:
Single costs ( for License): 100$
Maintenance costs etc. : 200 Year Yearly

Also an option that calculates the end date of the software would be nice. 
Now i have to put it manually everytime. A select Button thats add months/years would be nice

**Comment by Blackl1st** [comment]
+1 Would love to have this. We have a lot of student users and it would be nice to include year of graduation and state identifiers as custom fields.

**Comment by andreasoc** [comment]
yes thanks

**Comment by Metal-Frog** [comment]
+1
Custom fields for accessories and components! Urgently!

**Comment by sjlaird** [comment]
Maybe these requirements could be broken down again with the hope of some features being introduced?

I a really feeling the pain of not being and to handle renewals of maintained licenses well.

**Comment by lovettbp** [comment]
Please implement this! We use Dynamics and being able to track security roles, units, etc. with custom fields is absolutely vital. we are stuck with a spreadsheet until this is properly implemented.

**Comment by fyberoptik** [comment]
Yes please!

**Comment by KrishnaprabhaSNaik** [comment]
Please add custom fields to components.

**Comment by erinFTW** [comment]
+1 for sure!

We have many fields that are pertinent to the location that are not a part of the standard location fields. Having the ability to implement this would be a huge boon for our use.

**Comment by jcschurdell** [comment]
+1 Custom Fields would be the final addition we need to fully implement Snipe-IT at my company. Especially for consumables so it is easier to track why they were used.

**Comment by sinichi19** [comment]
Looking forward for this +1

**Comment by PhantomPhreak** [comment]
+1 for custom fields for components

**Comment by igloo777** [comment]
+1 for custom fields for components, it would be a big step forward

**Comment by viclou** [comment]
+1 FD#9212

**Comment by stbc** [comment]
+1

**Comment by johnrgithub** [comment]
+1. Custom fields for licences please - any update on timeline?

**Comment by acerspyro** [comment]
+1, licensing a weird software like WriteUp with dynamic keys and hardware fingerprints is near impossible at the moment.

**Comment by ivan5rov** [comment]
+1 Custom fields in Software will be very useful

**Comment by vidkun** [comment]
Another vote for needing custom fields on software/licenses. This is critical for us in choosing a replacement for our current system.

**Comment by gideonpelser** [comment]
+1 Custom fields for Components please

**Comment by eeyepee** [comment]
Custom fields on all non-assets would be great to have in this system. Components and consumables need more fields, as an example, checkout consumables to asset. Other use case would be having model and notes on components and consumables.

**Comment by happicamper** [comment]
+1 for this.
-1 for stale bot.

**Comment by aimerbois** [comment]
Custom fields for Non-Assets would be very useful to have.

**Comment by anmol26s** [comment]
+1

**Comment by JJLosier** [comment]
+1 custom fields across all non-assets - 3 years since the initial request?

**Comment by snipe** [comment]
> 3 years since the initial request?

The initial request was made by *me*, the maintainer. And yes, it's been three years, thousands of commits and hundreds of features+fixes later.

**Comment by JJLosier** [comment]
> > 3 years since the initial request?
> 
> The initial request was made by _me_, the maintainer. And yes, it's been three years, thousands of commits and hundreds of features+fixes later.

I don't want to take up the thread, but well played :) I just figured another ticket wasn't necessary since it was already opened.  I believe you when you say thousands of code commits.. etc.  I also don't know what kind of crazy feature request queue you have.  I think I've looked for this before for Software.  Do we know if/when it'll be available? :D

**Comment by sjlaird** [comment]
So is it about time we give up on this and look for something else?

-------------------
Simon Laird
Serco for the European Space Agency
Tel: +31715658756

> On 8 May 2019, at 20:56, snipe <notifications@github.com> wrote:
> 
> 3 years since the initial request?
> 
> The initial request was made by me, the maintainer. And yes, it's been three years, thousands of commits and hundreds of features+fixes later.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.

This message is intended only for the recipient(s) named above. It may contain proprietary information and/or
protected content. Any unauthorised disclosure, use, retention or dissemination is prohibited. If you have received
this e-mail in error, please notify the sender immediately. ESA applies appropriate organisational measures to protect
personal data, in case of data privacy queries, please contact the ESA Data Protection Officer (dpo@esa.int).

**Comment by snipe** [comment]
> Do we know if/when it'll be available? :D

Definitely will be. Not sure about when. We have to ship v5 before we're going to look at this issue, as it affects a *lot* of code, and changes some of the fundamental ways we approach items/models/etc.

**Comment by snipe** [comment]
> So is it about time we give up on this and look for something else?

You can see how there's no way I can answer that for you, right? 

If it's absolutely critical that you have this feature *today*, then yeah, you probably should give up on it and look for something else. 

If you can wait until we ship v5 and can look more closely at carefully implementing and testing the kind of large changes that this feature requires, then maybe not. 

At the end of the day, we want people to have a system - whatever system it is, even if it's not ours - that works for them. But we're also not going to just slap on something like this and hope for the best.

**Comment by SpearRIT** [comment]
I see that this is in the roadmap, but still adding a +1.

**Comment by syswpit** [comment]
Just started using Snipe it and immediately ran into this issue.
Apart from adding "custom" fields  to e.g. Accessories, I also want to add already-existing fields to Accessories, for example: Status. 

Why do some of these fields disappear? Why not leave them in the list and allow user to choose whether to use it or not?

**Comment by wilkems** [comment]
Just went to import licenses for Bluebeam and noticed my custom fields didn't show. That's when I found this article. Would very much enjoy this feature!

**Comment by kgortmak** [comment]
No one is talking about this one, but I really could use custom fields in users. Only 1 Phonenumber field?!? almost everyone has multiple number nowadays: office, mobile, soft (skype) phone, home... possibilties are endless, and most of the fields are available in AD for import.

**Comment by cliff-LINKS** [comment]
+1 for custom fields in users and other non-assets.

**Comment by gpproton** [comment]
Really hoping this will be done, we're currently stuck trying to add custom fields a license

**Comment by infectormp** [comment]
+1

**Comment by eeyepee** [comment]
+1

**Comment by pierinz** [comment]
+1

**Comment by createchange** [comment]
+1

**Comment by twsit988** [comment]
+1

**Comment by ninelciocoiu** [comment]
+1

**Comment by arcxander** [comment]
+1 for this

**Comment by nicosalvadore** [comment]
+1 for this also

**Comment by meduntsov** [comment]
+1

**Comment by level42ca** [comment]
Are we at all close to this?

**Comment by Paris999** [comment]
In the Licence Plan is also missing, that some special Software only working with Dongle. But I can only enter the Software Licence. But this Software Licence can be variat to the Number of dongle for this Software.
Please enter the possiblity that a Dongle and How many Dongle for a Software are available with the Software Licence. This where also a good choice, because some Software have Sharing Licences over Server with that Dongle entered.

**Comment by arcxander** [comment]
still waiting for this feature.

**Comment by k4d1sm0** [comment]
+1

**Comment by arcxander** [comment]
It's 2020 and we badly need this feature. I hope it will be implemented soon.

**Comment by niklasskoldmark** [comment]
+1

**Comment by mahiryildiz** [comment]
+1

**Comment by Speed1** [comment]
+1

**Comment by rostisluv** [comment]
+1

**Comment by logeshkanna** [comment]
+1

**Comment by dhooper33** [comment]
+1

**Comment by fbatsl** [comment]
+1

**Comment by phinguyen** [comment]
+1

**Comment by akarpilovsky** [comment]
Also need this. The lack of customizing and ridiculous number of "Mandatory" fields is really turning me off to the whole product. Why have so many mandatory fields? Why not just let us choose?

**Comment by shubham8agar** [comment]
+1 It is needed badly..Any ETA for this feature will be highly appreciated.

**Comment by Arapacana** [comment]
+99 -- this feature is BADLY needed to make this software much more robust and usable.  My example is that I need to track UoM (units of measure) and a special internal product code for consumables.  But this needs to be implemented system-wide, so that custom fields can be added to Licenses, Accessories, Consumables, Components, and People. 

I'm sure it's complicated to implement, but possible, and would add at least 20% to the use-case scenarios where it could be implemented smoothly, vs. trying to kludge something together like trying to track consumables as assets so you can utilize custom fields.

**Comment by bearkuma123** [comment]
+1

**Comment by wellmanntom** [comment]
+1 I really need to add things like architecture or language to my licenses!

**Comment by heikoboehme** [comment]
+1 .. urgently needed  
Thanks for this great tool :-)

**Comment by kreinartz83** [comment]
+1, I could absolutely use more custom fields in all the things.

**Comment by ReyoMediaCY** [comment]
+1

**Comment by mistamayor** [comment]
+1

**Comment by ColorsASD** [comment]
+1

**Comment by lee-skateistan** [comment]
+1

**Comment by STI-IT** [comment]
+1 We are just now moving from excel to Snipe IT, having the ability to assign components to a asset and specify details such as in the instance of Ram, like dimm or so-dimm, ram size in GB, type like DDR4. This would also help with storage media tracking like hdds in which when the component is assigned to the machine you can add a unique asset tag. Now that does get into the issue of the difference between components and assets but I know many companies that track the individual components aswell as the usable machine, but the difference is the asset is assigned to a department or user while the components will only be assigned to asset itself and is of no use unless it is installed in the asset.

**Comment by johnbchron** [comment]
+1. We all really appreciate your work, just need some things to be more systematic before we can use it in production :)

**Comment by Forefallen** [comment]
+1 Having custom fields in /licenses would make my life much simpler as someone who uses the api almost exclusively for displaying asset data and approved software to people that don't have access to snipe-it within the organisation.

Thanks for your work!

**Comment by ryanmerolle** [comment]
+1 for custom fields to locations.  Items I would look to track, using custom fields, for a location would include:
- Contact details
- Website portals
- Landlord

**Comment by simonshtipelman** [comment]
+1

**Comment by yosiasz** [comment]
https://www.youtube.com/watch?v=IrCEhRNgGHY

give me give me, i need i need i need

what about Bob

<blockquote><img src="https://i.ytimg.com/vi/IrCEhRNgGHY/hqdefault.jpg" width="48" align="right"><div><img src="https://www.youtube.com/s/desktop/4aac258c/img/favicon_144.png" height="14"> YouTube</div><div><strong><a href="https://www.youtube.com/watch?v=IrCEhRNgGHY">Gimme Gimme, I Need I Need..."What About Bob?"</a></strong></div><div>Bob Wiley begs for an impromptu psychiatric session from Dr. Leo Marvin while he's on vacation in New Hampshire.</div></blockquote>

**Comment by snipe** [comment]
It's coming, it's coming

**Comment by TinaMccray** [comment]
+1 Need this

**Comment by ColorsASD** [comment]
+1

**Comment by arcxander** [comment]
> It's coming, it's coming

thank you snipe! looking forward to this

**Comment by TinaMccray** [comment]
Here is the link if you want to see the ticket
https://github.com/snipe/snipe-it/issues/4261#issuecomment-700611362

Tina McCray
Computer Technician
Salem Academy and College
601 S. Church Street
Winston-Salem, NC 27101
Fine Arts Center, Room 310

**Comment by creactiv** [comment]
+1 Since Software Licenses are typically bound to software versions i wanted to add a custom field to licenses, which wouldn‘t work... Same for license type GPL, Commercial, personal with up to  x devices and the like...

So any news on this one?

**Comment by hyggja** [comment]
+1

**Comment by andro44** [comment]
+1

**Comment by Relix360** [comment]
+1

**Comment by yosiasz** [comment]
what would it take to make this happen? bounty?

**Comment by snipe** [comment]
@yosiasz It would require a huge rewrite, and I'm not even sure we'd take a PR for it since it touches so many things. It's in the works. Please just remain patient.

**Comment by viclou** [comment]
+1 FD 19378

**Comment by sowi-it** [comment]
+1

**Comment by winstan** [comment]
Looking forward to this. Intend to track multiple network interfaces (with extended details) per asset. :)

**Comment by MrCaspan** [comment]
I would love to donate to the cause also Snipe if that helps with getting the ball rolling after v5 is out!

I know this is a no no but for now we will never use the 'Licensed to Email' or 'Licensed to Name' field in the Software form, I just renamed the language value for it in the language file (snipe-it/resources/lang/en/admin/licenses/form.php) to the field name I wanted. I have to redo this change after each update but if you only need one or two fields for now is a nice simple easy stop gap. For us we need just an single Media ID field that tell us where the Media is located on our filling system for this software.

Also just an idea and maybe temporarily Snipe could use some JavaScript and pull JSON data from the notes field and dynamically show it on the table view? Just an idea

**Comment by kirillk77** [comment]
+1

**Comment by crax0** [comment]
+1

**Comment by alainseys** [comment]
+1

**Comment by avigi25** [comment]
+1

**Comment by SufiSR** [comment]
+1 Custom Fields for Licenses
I had this wonderful idea of tracking GDPR compliance for software licenses / products where i would have custom fields for the GDPR policy review data, data residency and to what extend a user may use the software in terms of PII - which I would then inform them about in the checkout email. 

Too bad it's not there yet.

**Comment by snipe** [comment]
Too bad indeed.

**Comment by MrCaspan** [comment]
@snipe Just lock this thread. You have told us it's coming but people want to still throw the team shade and constantly saying +1 You have better things to do then to listen to the same story over and over.

**Comment by acerspyro** [comment]
@MrCaspan I'm a bit disillusioned with SnipeIt at this point. I thought them incorporating into a company that funds itself off of providing support would allow them to make development faster, but most of what they do right now seems to be code maintenance rather than development.

The fact remains that they are still the best open-source asset tracker out there, but sometimes being the best in a category just isn't enough when crucial features are missing and lots of issues remain unsolved. Our company is now at the point where we're looking for alternatives as we've just hit too many limitations and had too many issues with the software.

**Comment by MrCaspan** [comment]
This is an issue tracing system. The issue has been described and the team has replied it's in the works but it's going to take a lot to do! Emotions are now in play and people feel frustrated understandably but. It's free software. If it does not suit your needs and the development cycle is not fast enough then you only have one choice. Being frustrated at the developers won't do a thing but make them feel like why bother even trying, they have said it requires a full rewrite to do this. If you can't wait and you have to have these features then you have to do what you have to do. They have a very thankless job and are always expected to produce! I get people's frustration but let's give them some breathing room and some encouragement!!

**Comment by acerspyro** [comment]
> This is an issue tracing system. The issue has been described and the team has replied it's in the works but it's going to take a lot to do! Emotions are now in play and people feel frustrated understandably but. It's free software. If it does not suit your needs and the development cycle is not fast enough then you only have one choice. Being frustrated at the developers won't do a thing but make them feel like why bother even trying, they have said it requires a full rewrite to do this. If you can't wait and you have to have these features then you have to do what you have to do. They have a very thankless job and are always expected to produce! I get people's frustration but let's give them some breathing room and some encouragement!!

I'm not throwing shade at the team, I'm simply stating that even though the team is receiving funding through paid support and hosting, the development is stuck in a maintenance cycle with little improvements.

It's important, even as an open-source developer, to hear out your users and understand how they feel about the software you're building so you don't just end up going in a random direction and stay on track with the needs of the users. Issue trackers are as much about bug reports and feature requests as they are about discussions regarding the project itself.

I'm glad people were able to take personal time out of their own day to come together and build this software. It's served us well, I simply believe our company has hit its limitations and outgrown it.

If the team has no desire to work on this feature, they should lock and close the thread.

Otherwise, it should stay open, and I recommend that people instead voice their support using a 👍 reaction on the original request so as not to spam the conversation.

**Comment by snipe** [comment]
@acerspyro You can throw all the shade you like, it won’t change the fact that there’s a lot going on behind the scenes, including the planned release of v6 in September. The feature is being worked on actively, but we are a small team. V6 required an entire framework upgrade and is being worked on every day, and custom fields have always been a hard problem to solve correctly. (Easy to solve, not easy to solve correctly. There are speed implications, database limitations, UX and UI implications, and so on - not to mention the dozen other features we’re trying to put into v6.)

If you need to move to another solution, be my guest, but please don’t lecture me on how to run an open source project or my company.

**Comment by snipe** [comment]
@MrCaspan thank you for your patience and understanding. This is still very much a top priority for the next major release. ❤️

**Comment by rhelbling22** [comment]
Following

**Comment by rhelbling22** [comment]
Adding custom fields and fieldsets to Accessories, Consumables and Components is definitely a need.  Do you have any sort of an ETA on this feature, please?  Thanks.

**Comment by clintseban** [comment]
@snipe Any progress on this requirement? Many of them are eagerly waiting to get an answer from the Snipe-IT team.

**Comment by yechunshen** [comment]
already done?

**Comment by ryanmerolle** [comment]
> already done?

Not sure what you are talking about.  The issue is still open.  If it’s a sarcastic comment, @snipe and team do an amazing job and it will come when it comes.

**Comment by snipe** [comment]
It's still on the short term roadmap, but it affects a LOT of moving parts, and we have to be careful when we make such sweeping changes. There are *always* unintended consequences.

**Comment by nghia-dang** [comment]
From what I could tell, people had asked for custom fields that can be added to people, but does this include a custom field added to an asset, whereby you can select via a picker to assign a person to the custom field within the asset? 

Eg. If I had a 'Purchaser' custom field on an asset that I could choose a person from (and if I visited the person details I'd see the assets assigned via this field)

**Comment by Noofayroz** [comment]
is it already there ?

**Comment by jbryant88** [comment]
Im just wondering if the Custom Fields for Licenses is something that is coming? OR if its possible to add to the Purchase costs to have a separate entry for the Total cost of the Licenses (Ie Office with 100 seats) and separate the per seat cost that posts to the People

**Comment by daliborpuljiz** [comment]
I've come here just to find out the custom fields are not ready for non-assets. Looking forward to it, as well :)

**Comment by rmscode** [comment]
Bump...PLEASE. Custom fields for all categories.

**Comment by ZebboKlaufix** [comment]
This would be a great thing to have. Please allow custom fields for categories.

**Comment by PhantomPhreak** [comment]
This feature was also requested here in 2016: https://github.com/snipe/snipe-it/issues/2934
That thread contain a lot of interesting details, as well as signs of how violent the repository owner’s reaction was to users’ attempts to explain how important this feature is to them.
And after 7 years, nothing has changed.
I wish everyone waiting for this feature strong patience.

**Comment by SufiSR** [comment]
With all due respect - I am not sure how this last post is any helpful. First and foremost, it’s an open source project and actually a very good one - and I would think that might be due to the repository owner.

**Comment by kgortmak** [comment]
Agreed. Read the whole long discussion, and it has nothing to do wit the request here. Basic problem there is that the dev's should not have used the fieldname 'serial number' in the components data set. In stead they shoud have used 'part number', as that would identify a specific part (or series of parts with the same part number) Calling it Serial number (which basically is a unique identity number) causes people to create expectations that are not correct.  Other thing in that discussion is the fact a lot of people completely ignore the fact that you can check out assets to another assets. Or create a specific category. There where people calling switches components.  In my book, switches are assets in the Network Infrastructure category. You can use custom fields to register stack position or something.

Nevertheless, these kind of feature requests are not "must have's" according to the dev's, which is a pitty, since so many people acknowledged the need for it.  ;-)

**Comment by iamsumesh** [comment]
+1.

**Comment by MrCaspan** [comment]
I think it's time to lock the thread. No one knows how to read

**Comment by heikoboehme** [comment]
just as an idea: have a look at the custom field / value function as implemented in redmine  ( redmine.org) . it works incredibly well. custom fields have been implemented for all the different documentation objects in a very simple way

**Comment by heikoboehme** [comment]
just as an idea: have a look at the custom field / value function as implemented in redmine ( redmine.org) . it works incredibly well. custom fields have been implemented for all the different documentation objects in a very simple way

**Comment by rmscode** [comment]
> I think it's time to lock the thread. No one knows how to read

What exactly are we missing?

**Comment by SufiSR** [comment]
> > I think it's time to lock the thread. No one knows how to read
> 
> What exactly are we missing?

Nothing really - the other guy seems to be misreading the duplicate ;)

**Comment by MrCaspan** [comment]
Sorry I think 2 different issues have been merged and the huge thread is gone. You can follow an issue without pinging the entire community to follow this issues. It's annoying to get a "me too ping" every week. This was in the old thread to just subscribe if you want updates.. the +1 comments were just annoying and not helpful. The dev knows it's an issue and have said it's a full rewrite to do it so maybe happening but maybe not because it's huge!! Subscribe to the issue and wait patiently or pitch in to help with the project. No more +1's please or only post if you have info about it.. That was the summary of the old discussion about this. Just my 2 cents take it for what it is..

**Comment by rmscode** [comment]
> It's annoying to get a "me too ping" every week. 

Fair point. 

I haven't spent a great amount of time reading through all the related issues, but what I got out of it is that Snipe doesn't seem all that interested considering the fact that there are "hundreds" of other features requests on the to-do list, allegedly. I recall reading one of her direct replies to someone being a bit snarky . . . "I welcome your pull request". Snipe's [most recent solution]( https://github.com/snipe/snipe-it/issues/2934#issuecomment-1485467543) is almost as if she's saying "deal with it". I could be reading her tone completely wrong. 

I guess I'll just patiently wait and remain hopeful.

**Comment by JCollinsMPS** [comment]
I, too, look forward to expanded functionality. We need to add employee keycards into our system and would like to do so as accessories. However, there is no good way to display the "token #" on the keycard in Snipe. Using the notes section when checking it out to a user only shows that information in the history of the accessory, not the user. We will probably just make it an accessory at this point so we can track things in Snipe.

**Comment by rthreis404** [comment]
Hello Everyone,
I'm very interested in this feature too.
I'd like to be able to define things like :

Assets:
- Laptops
- Desktop
- ...
Accessories:
- Keyboard
- Headset
- Mouse
- Docking station

Custom Fields
- Connection : wired || wireless
- Layout : AZERTY (BE) || AZERTY (FR) || QWERTZ (DE) || QWERTY (UK)
- Power : 60W || 120W || 180W ||

Be able to:
Layout used with: Asset>Laptops && Accessories>Keyboard
Connection used with: Accessories>Keyboard && Headset && Mouse
Power used with: Accessories>Docking station

I'm not sure if this is clear, and I don't know how the database tables are linked together in order to set this up.
I would also like to thank you for the great work you're doing on this.
Have a great day all,

**Comment by 00shorty** [comment]
We're also interested in this feature.

**Comment by mai-wand** [comment]
Custom fields for accessories would be great and very necessary.

@snipe is there a roadmap?

**Comment by koyuncuoglugithub** [comment]
up

**Comment by kekec777** [comment]
Sign me up ! :)

**Comment by rmscode** [comment]
> Custom fields for accessories would be great and very necessary.
> 
> @snipe is there a roadmap?

I don't believe it's on any "roadmap". I do recall her replying to a user once with, "I welcome your pull request" . . . as if to say "If you want this feature, contribute to the project and add it yourself."

**Comment by gertz22** [comment]
UP
I'm also interested in this feature mainly licences.

**Comment by turist79** [comment]
I don`t think this will be ever implemented considering for how long the issue is being discussed.
The lack of custom fileds in components and inability to assign multiple fieldsets to one asset (component) are showstoppers for us unfortunately.
I wish there existed blend between SnipeIt and Openboxes/PartKeepr... :)

**Comment by snipe** [comment]
> I don`t think this will be ever implemented considering for how long the issue is being discussed.

@turist79 - appreciate your unsolicited opinion, but that's simply not true. It's a massive change, and we have to be careful. There's already a PR open for adding custom fields to users, but we have to wait for v7 to be released before we can move forward on that.

**Comment by turist79** [comment]
> > I don`t think this will be ever implemented considering for how long the issue is being discussed.
> 
> @turist79 - appreciate your unsolicited opinion, but that's simply not true. It's a massive change, and we have to be careful. There's already a PR open for adding custom fields to users, but we have to wait for v7 to be released before we can move forward on that.

Happy to hear that there is some progress in that regard.
Probably the issue here could be lack of feedback to the community since the feature is so long-awaited by many.
PS: in my opinion the best solution for those who miss this feature would be to use assets instead of components, the possibility to "link" assets to assets will create same kind of relationship actually but with added benefits of custom fields.

**Comment by snipe** [comment]
> Probably the issue here could be lack of feedback to the community since the feature is so long-awaited by many.

The ticket itself shows multiple updates in the way of linked issues. There just isn't that much tangible to report right now, as the underpinnings need to be updated to v7, and then the new branch needs to be tested on v7, but the UI updates that this would require (and 100 other things) are dependent the v7 release, as we've mentioned in those linked tickets, on our blog, in our Discord, and every other place imaginable. 

> PS: in my opinion the best solution for those who miss this feature would be to use assets instead of components,

This is literally what we always recommend.

>  the possibility to "link" assets to assets

You can already assign assets to other assets.

**Comment by mauro-miatello** [comment]
> the possibility to "link" assets to assets will create same kind of relationship actually but with added benefits of custom fields.

https://github.com/snipe/snipe-it/issues/14537

**Comment by PietSkietVerniet** [comment]
I'm very keen on seeing a feature that lets me add custom fields to user accounts ("people") in Snipe-IT.
Right now, I'm adding a tag in the Notes section which i can query from the database directly to do a report on.

**Comment by jay-chuang** [comment]
When these request will be added into Snipe-it system?

**Comment by huifei0129** [comment]
I'd love to have custom fields for non-assets also.
Hope it can be done

**Comment by shanemartin22** [comment]
+1 for custom fields in Locations and People.
Now that v7 is out, any expected timeframe for this?

**Comment by snipe** [comment]
There is a WIP PR already open for custom fields on users.

**Comment by deeepakkumarthakur16** [comment]
Much awaited feature.

**Comment by ZapodD** [comment]
Any news, any progress information?

**Comment by denladdi** [comment]
very interested in this feature too!

**Comment by jLebioda** [comment]
+1

**Comment by phgravel** [comment]
Also interested!

**Comment by Mike-NP** [comment]
+1

**Comment by laserlars3000** [comment]
+1, in combination with fieldsets would be great

**Comment by Homeoffice74** [comment]
+1,  Unfortunately, license management currently lacks too much information to be able to use it productively. E.g. Are these concurrent or user licenses, what maintenance costs are incurred, what are the cancellation options for the license or maintenance, is there software assurance, are contracts automatically extended annually, is there a minimum contract term, etc etc

**Comment by luisffl01** [comment]
+1

**Comment by l-andrad** [comment]
+1, custom fields throughout the system would be great!

**Comment by Callandt** [comment]
+1

**Comment by 00shorty** [comment]
+1

**Comment by tianji1993** [comment]
+1

**Comment by mrzasranec** [comment]
+1

**Comment by rodrigobuttow** [comment]
+1

**Comment by rpolaskii** [comment]
+1

**Comment by trebogates** [comment]
Would love the ability to add custom/additional fields under Companies - mainly so I can use an abbreviated Company Name on Asset Labels

**Comment by yasinvs** [comment]
+1

**Comment by Haelgomatic** [comment]
+1    for Location. Whats the roadmap? Is there a way to help you?

**Comment by ricardomeleiro** [comment]
+1

**Comment by mcarasso-PPD** [comment]
+1

**Comment by rpolaskii** [comment]
+1

**Comment by ce603** [comment]
+1

**Comment by patstevens4** [comment]
+1

**Comment by TreeBranches** [comment]
+1 for adding Sage reference numbers to Licences.

**Comment by boe-fmb** [comment]
+1

**Comment by jcohenacuity** [comment]
+1

**Comment by it-management-stadtbetrieb** [comment]
+1

**Comment by rpolaskii** [comment]
+1

**Comment by thisisnotgroiss** [comment]
+1

**Comment by yasinvs** [comment]
+1

**Comment by SageDearborn** [comment]
+1, especially for licenses and users.

@snipe, bountysource appears to be dead.  Have you migrated to a new bounty platform? Can you provide any insight on how much it would cost to get this feature developed?

**Comment by snipe** [comment]
Bountysource has been dead for years now. 😞 But also this feature is way more work than the bounty was for. We’ve got a PR in the works for custom fields for users, since that’s the hardest to execute (they have some fields and behaviors that are different, since they can’t be checked out, etc.) Still very much a priority for us, we just have to make sure we do it right the first time.

**Comment by SageDearborn** [comment]
Thanks snipe. We make a lot of use of your software. We are very grateful
for the open-source nature. If you come up with a way we can support new
feature development, please feel free to tag me and I will see if I can
talk my management into sending money your way.

With Love and Light,
Sage
Server/Network Analyst
Dearborn Public Schools

**Comment by it-management-stadtbetrieb** [comment]
+1

Hi, thanks a lot for all your great work on Snipe-IT! 🙌
Just wondering if this feature is still on the roadmap, and if you might have a rough idea of when it could be available. It would be super helpful for us (e.g. for cost centers).

Thanks again!

**Comment by snipe** [comment]
@it-management-stadtbetrieb Most definitely is, yep!

**Comment by jbagot** [comment]
+1

**Comment by MikeT300** [comment]
+1 here too.

We're trying a workaround through Assets to mark certain software as mission-critical. It's admittedly pretty clunky though, and the ability to add custom fields in Licenses would be perfect.

**Comment by DerKrisp** [comment]
+1

**Comment by yasinvs** [comment]
+1

**Comment by viclou** [comment]
[FD-54583] for locations

**Comment by mai-wand** [comment]
Accessories need definitely custom fields

**Comment by viclou** [comment]
[FD-55180]

**Comment by tluckhardt-alvys** [comment]
+1 - want to add custom fields to licenses, and also support a custom field for "role" on the license checkout.

**Comment by maxusmaus** [comment]
+1

**Comment by mai-wand** [comment]
![image](https://github.com/user-attachments/assets/9dede3ae-e1bd-49c6-a299-d3bf7815b1c7)

**Comment by l-andrad** [comment]
We need this for licenses internally, so we scoped it out against v8.7.2. We were going to
  propose a design, then went looking for prior art first — and the design already exists. So
  this is a question about what would actually be useful, not a proposal.

  For anyone else reading this thread: the work is much further along than it looks from here.
  #16100 isn't users-only. It carries the generic mechanism — a type discriminator on both
  custom_fields and custom_fieldsets (migrations dated 2021-06-29), HasCustomFields with
  getFieldset() / getFieldsetKey() as the override point so each item type can resolve its own
  container, plus generalized default values. Users are just the first consumer, and the
  hardest one, which matches what @snipe has said here repeatedly.

  What seems to be blocking it isn't the design and isn't willingness. It's that the branch has
  been rebased across several major versions and is now carrying an enormous amount of merge
  noise, and its author is understandably busy elsewhere. #19098 looks like an attempt to break
  the trait out of it and land that piece alone — opened and closed the same day, no comment,
  and app/Models/Traits/HasCustomFields.php still isn't in v8.7.2.

  So, two questions:

  1. Why was #19098 closed? If the extraction hit something real, we'd rather know than
     rediscover it. If it was just bandwidth, that changes what we offer.
  2. Would a decomposition of #16100 be welcome, or would it step on toes? What we have time to
     do is carry @uberbrady's design forward without changing it — same FQCN type
     discriminator, same getFieldsetKey() override — re-landed against current develop as
     reviewable slices instead of one branch. Roughly: schema + trait + tests with no behavior
     change first, then users, then the rest. We'd want him in the loop on it, not around him.

  One thing we can contribute on top rather than just port: licenses, accessories, consumables
  and components all already carry category_id, and categories.category_type is already
  constrained to asset|accessory|consumable|component|license (app/Models/Category.php:51). So
  a nullable categories.fieldset_id gives those four a container for free — one
  getFieldsetKey() override, mirroring models.fieldset_id. It doesn't cover locations, models
  or maintenances, which several of the issues linked here also ask for; those each need their
  own override, which is exactly what that method is for.

  We won't open anything without a go-ahead. Just tell us which of the above is genuinely
  useful, and we'll do that one.
